src/inference/src/dev/core_impl.cpp

```diff
@@ -1077,8 +1077,14 @@ std::shared_ptr<const ov::Model> ov::CoreImpl::apply_auto_batching(const std::sh
         auto pos = deviceName.find_first_of(":");
         if (pos == std::string::npos)
             return model;  // BATCH device is already configured via the config
-        deviceNameWithBatchSize = deviceName.substr(pos + 1);
+
```

Comment:
Don't remove it. It is required by the next line and https://github.com/openvinotoolkit/openvino/pull/19541/files#diff-8d13c3a18a5f54a062387b01b0e2bb6dedc35c42e73c752ae00b73ee4c60f0fbR1136